File: Model/cnn.py
import tensorflow as tf
from tensorflow.python.keras.models import Sequential
from tensorflow.python.keras.layers import Dense, Dropout

def create_model(input_shape):
    model = Sequential([
        Dense(128, input_shape=(input_shape,), activation='relu'),  # Input layer
        Dropout(0.3),  # Dropout layer for regularization
        Dense(64, activation='relu'),  # Hidden layer
        Dropout(0.3),  # Another dropout layer
        Dense(2, activation='softmax')  # Output layer with 2 classes ('standing' and 'squatting')
    ])

    # Compile model
    model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])
    
    return model

File: Model/cnn.js
// This class defines the CNN model

import * as tf from '@tensorflow/tfjs';

export function createModel(inputShape) {
    const model = tf.sequential();

    // First convolutional layer
    model.add(tf.layers.conv2d({
        inputShape: inputShape,
        filters: 32,
        kernelSize: 3,
        activation: 'relu',
        kernelRegularizer: tf.regularizers.l2({ l2: 0.01 }) // Add L2 regularization
    }));

    // Add L2 regularization to all convolutional and dense layers
    model.add(tf.layers.conv2d({
        filters: 64,
        kernelSize: 3,
        activation: 'relu',
        kernelRegularizer: tf.regularizers.l2({ l2: 0.01 })
    }));
    
    model.add(tf.layers.flatten());

    // Dense layer with L2 regularization
    model.add(tf.layers.dense({
        units: 128,
        activation: 'relu',
        kernelRegularizer: tf.regularizers.l2({ l2: 0.01 })
    }));

    // Output layer
    model.add(tf.layers.dense({
        units: 1,
        activation: 'sigmoid'
    }));

    return model;
}